include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,285 @@
+            for (std::size_t kernel_row = 0; kernel_row < kernel.size(); ++kernel_row)
+            {
+                flip_ker_row = kernel.size() - 1 - kernel_row;      // row index of flipped kernel
+
+                for (std::size_t kernel_col = 0; kernel_col < kernel.size(); ++kernel_col)
+                {
+                    flip_ker_col = kernel.size() - 1 - kernel_col; // column index of flipped kernel
+
+                    // index of input signal, used for checking boundary
+                    row_boundary = view_row + (kernel.center_y() - flip_ker_row);
+                    col_boundary = view_col + (kernel.center_x() - flip_ker_col);
+
+                    // ignore input samples which are out of bound
+                    if (row_boundary >= 0 && row_boundary < src_view.height() &&
+                        col_boundary >= 0 && col_boundary < src_view.width())
+                    {
+                        // We ensure that we consider only those pixels which are overlapped on a
+                        // non-zero kernel_element as
+                        if(kernel.at(flip_ker_row,flip_ker_col))
+                        {
+                            if(identifier == boost::gil::morphological_operations::dilation)
+                            {
+                                if(src_view(col_boundary, row_boundary) > target_element)
+                                    target_element = src_view(col_boundary, row_boundary);
+                            }
+                            else if(identifier == boost::gil::morphological_operations::erosion)
+                            {
+                                if(src_view(col_boundary, row_boundary) < target_element)
+                                    target_element = src_view(col_boundary, row_boundary);
+                            }
+                        }
+                    }
+                }
+            }
+            dst_view(view_col, view_row) = target_element;
+        }
+    }
+}
+
+/// \brief Checks feasibility of the desired operation and passes parameter values to the
+/// function morph_impl alongwith individual channel views of the input image.
+/// \param src_view - Source/Input image view.
+/// \param dst_view - View which stores the final result of operations performed by this function.
+/// \param kernel - Kernel matrix/structuring element containing 0's and 1's which will be used for applying the required morphological operation.
+/// \param identifier - Indicates the type of morphological operation to be applied.
+/// \tparam SrcView type of source image.
+/// \tparam DstView type of output image.
+/// \tparam Kernel type of structuring element.
+template <typename SrcView, typename DstView,typename Kernel>
+void morph(SrcView const& src_view, DstView & dst_view,Kernel const& ker_mat, 
+                                                            morphological_operations identifier)
+{
+    using namespace boost::gil;
```

Comment:
What is this for?

Reply:
In an earlier version of this file, I was using ```gray8_image_t``` , I changed that but forgot to remove the namespace part,apologies .